selfhost/codegen.jakt

```diff
@@ -248,6 +248,7 @@ struct CodeGenerator {
         )
         mut output = ""
         output += "#include <lib.h>\n"
+        output += "#ifdef WINDOWS\n#include <Windows.h>\n#endif\n"
```

Comment:
This seems to be a reasonable fix... except that Windows.h is very, very heavy. 

I suspect that this will fail in CI because one of the headers pulled in by Windows.h  has a `#define` for `Yield`, which is used as a type in the compiler, and we didn't explicitly `#define NOMINMAX` here, so `min` and `max` will be macros in any generated code. 

I did quite the dance for this in #1149 . One of the main motivations to use a separate library for the runtime was to shield normal jakt code from platform-specifics and awkward headers. Side note, `<vxWorks.h>` defines `OK` and `ERROR` as integer constants :).

From what i've seen, conventional windows wisdom would be to actually forward declare the `SetConsoleOutputCP()` method the way it shows up in the Windows SDK headers. Without swapping over to my windows box, something like `extern "C" __cdecl void SetConsoleOutputCP(DWORD code_page);`

Reply:
hm. I might be wrong about the compile failure? 

Unless `WINDOWS` isn't actually defined? I thought the OS detection macro was normally `_WIN32`

Reply:
@ADKaster not sure but it did work locally with WINDOWS. I'll move it to `_WIN32` and do the forward declare like you suggest

Reply:
thanks! side note on the forward declare, I think you can technically forward declare a global method at function scope. though that might be more awkward than forward declaring it up next to lib.h.

Not pulling in Windows.h also means either a local define or hard coded constant for CP_UTF8, which is `65001` iirc. 

Ah but you've already pushed it optimistically. :) The compiler shall be the source of truth on this one I think.